### PR TITLE
 *: Fix issue#1114

### LINKED ISSUE #1114: Set @tmp:=@tmp+1 failed
1. What did you do?
```
SET @tmp=0;
SET @tmp:=@tmp+1;
select @tmp;
```

2. What did you expect to see?

+------+
| @tmp |
+------+
|    1 |
+------+

3. What did you see instead?

ERROR 1105 (HY000): line 1 column 9 near ":"

### PR DESCRIPTION
**Pull request by zimulala**
fix #1114

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -1537,7 +1539,18 @@ NUM:
 	intLit
 
 Expression:
-	Expression logOr Expression %prec oror
+	"USER_VAR" assignmentEq Expression %prec assignmentEq
```
Comment (shenli) [inline parser/parser.y]: There will be conflict without "%prec"?

**Comment by shenli** [comment]
LGTM

File: parser/parser.y [hunk 2]
```diff
@@ -663,6 +664,7 @@ import (
 %left   tableRefPriority
 %precedence lowerThanOn
 %precedence on
+%left	assignmentEq
```
Comment (zxylvlp) [inline parser/parser.y]: The assign op should be right instead of left. But the rule make it unnecessary. I think we should keep it.

**Comment by zxylvlp** [comment]
LGTM

**Pull request merged**